data/lua/carryover_gold.lua

```diff
@@ -0,0 +1,248 @@
+local _ = wesnoth.textdomain "wesnoth"
+local carryover = {}
+
+--- Returns true iff there is not a single side that is controlled by the local human player.
+---@return boolean
+function carryover.get_is_observer()
+	for _, side in ipairs(wesnoth.sides) do
+		if side.controller == "human" and side.is_local then
+			return false
+		end
+	end
+	return true
+end
+
+--- Show names when thre are multiple persistent teams.
+---@return integer
+function carryover.get_num_persistent_teams()
+	local res = 0
+	for _, side in ipairs(wesnoth.sides) do
+		if side.persistent then
+			res = res + 1
+		end
+	end
+	return res
+end
+
+--- Gets the number of villages on the map.
+---@return integer
+function carryover.get_num_villages()
+	return #(wesnoth.map.find{ gives_income = true })
+end
+
+--- Like tostring(num) but uses a prettier minus sign, to be used in the UI.
+---@return string
+function carryover.half_signed_value(num)
+	if num < 0 then
+		return "-" .. tostring(math.abs(num))
+	else
+		return tostring(num)
+	end
+	return tostring(num)
+end
+
+--- Calculates the amount of carryover gold.
+---@param side side
+---@param turns_left integer
+---@return number (finishing_bonus), number (finishing_bonus_per_turn)
+function carryover.calculate_finishing_bonus(side, turns_left)
+	local num_villages = carryover.get_num_villages()
+
+	local finishing_bonus_per_turn = num_villages * side.village_gold + side.base_income
+	local finishing_bonus = math.ceil(side.carryover_bonus * finishing_bonus_per_turn * turns_left)
+	return finishing_bonus, finishing_bonus_per_turn
+end
+
+---@return boolean
```

Comment:
Please add a comment, either with `#` at the end of this line or with `---` on a separate line above, that explains what `get_report_visible` does.